Re-skeletonize

jsf9k opened the pull request:
## 🗣 Description ##

This pull request re-skeletonizes this repository.

## 💭 Motivation and context ##

This repository had been started from a skeleton in the distant past, but it was woefully out of date and lacked even a Lineage configuration.

## 🧪 Testing ##

All relevant automated testing passes.  The JavaScript LGTM check fails because there is no JavaScript code in this repository.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

mcdonnnj committed:
Pin ansible-lint to <5 to avoid upgrading to the latest version

This update broke our CI configuration as apb dispatched runes failed. This is
documented in https://github.com/cisagov/skeleton-ansible-role/issues/69 and
until things have shaken out a bit more we will hold on the 4.x release(s).

mcdonnnj committed:
Revert ansible-lint version update

The v5 release of ansible-lint introduces breaking changes, so we are holding
off on updating until things are resolved. Conversation about this can be
tracked in https://github.com/cisagov/skeleton-ansible-role/issues/69.

mcdonnnj committed:
Merge pull request #69 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_beautysh_pre-commit_hook

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/add_beautysh_pre-commit_hook

Add the beautysh pre-commit hook

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #70 from cisagov/bug/pin_ansible-lint_under_v5

Pin ansible-lint Version Under the 5.0 Release

mcdonnnj committed:
Merge branch 'develop' into improvement/use_fqcn_for_builtin_modules

mcdonnnj committed:
Merge pull request #67 from cisagov/improvement/use_fqcn_for_builtin_modules

Use Fully Qualified Class Name (FQCN) for Ansible Builtin Modules

mcdonnnj committed:
Merge branch 'develop' into remove_ubuntu16_support

mcdonnnj committed:
Merge pull request #66 from cisagov/remove_ubuntu16_support

Remove Support for Ubuntu Xenial (16) Hosts

mcdonnnj committed:
Merge branch 'develop' into lineage/skeleton

mcdonnnj committed:
Merge pull request #71 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge github.com:cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #13 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Use Go 1.16 in GitHub Actions workflow

With the release of terraform-docs v0.12.0 it now requires Go 1.16.

mcdonnnj committed:
Merge pull request #17 from cisagov/bug/use_go_1.16_for_terraform-docs_v0.12.0

Use Go 1.16 in GitHub Actions Workflow

felddy committed:
Improve gitignore file comments and organization.

* Add a comment describing what the files does.
* Add sections to organize patterns.

mcdonnnj committed:
Update pre-commit hooks

Hooks updated with the `pre-commit autoupdate` command.

Note:
The `ansible-lint` hook is intentionally being held back on 4.3.7 because of
ongoing issues with the 5.x version and how we use Ansible (standalone Galaxy
roles).

mcdonnnj committed:
Remove user from CODEOWNERS

Due to her departure this removes hillaryj from the default CODEOWNERS we use
in our projects.

dependabot[bot] committed:
Update ansible requirement from <3,>=2.10 to >=2.10,<5

Updates the requirements on [ansible](https://github.com/ansible/ansible) to permit the latest version.
- [Release notes](https://github.com/ansible/ansible/releases)
- [Commits](https://github.com/ansible/ansible/commits)

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Update configuration instructions for pyenv

With the release of pyenv v2.0.0 there is a breaking change around startup
logic that necessitates an update for our pyenv setup instructions. Also
add a statement about how to get configuration instructions from pyenv
itself.

mcdonnnj committed:
Merge pull request #71 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into maintenance/remove_user_from_codeowners

mcdonnnj committed:
Merge pull request #72 from cisagov/maintenance/remove_user_from_codeowners

Remove User from CODEOWNERS

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge branch 'develop' into documentation/update_pyenv_instructions

mcdonnnj committed:
Merge pull request #75 from cisagov/documentation/update_pyenv_instructions

Update Instructions for Configuring pyenv

mcdonnnj committed:
Merge branch 'develop' into improvement/gitignore_me

mcdonnnj committed:
Merge pull request #70 from cisagov/improvement/gitignore_me

Improve gitignore file comments and organization.

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #76 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

jsf9k committed:
Add default_tags to providers and alphabetize provider keys

jsf9k committed:
Require a version of the AWS provider sufficient to support default_tags

jsf9k committed:
Remove tags parameter from cisagov/molecule-iam-user-tf-module config

jsf9k committed:
Alphabetize remote state resource keys

jsf9k committed:
Separate TF .gitignore entries into their own section

mcdonnnj committed:
Standardize the layout of the Lineage configuration file

jsf9k committed:
Merge pull request #19 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

jsf9k committed:
Fix repeated typo/syntax error in providers.tf

jsf9k committed:
Merge pull request #20 from cisagov/bugfix/fix-typo

Fix repeated typo/syntax error

mcdonnnj committed:
Add style enforcement rules

Add rules to enforce ATX-closed headers, dashes for unordered list elements,
and `1.` for ordered list elements.

mcdonnnj committed:
Add rule for image headers

Add <h1> and <img> tags to the allowed list for MD033 (HTML elements) to
support using an image as the first thing in a markdown file (header image).

mcdonnnj committed:
Switch to a YAML markdownlint configuration file

This converts the existing `.mdl_config.json` file to an equivalent
`.mdl_config.yaml` file. The reference in the markdownlint pre-commit hook
configuration is updated to match.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add comments to markdownlint configuration

Now that this is a YAML file we can add comments explaining the rule
modifications we use. This will make it easier to edit or expand in the future.

mcdonnnj committed:
Add the validate_manifest hook from pre-commit

This hook will validate any pre-commit hook manifest files in the repository.

jsf9k committed:
Merge branch 'develop' into dependabot/pip/ansible-gte-2.10-and-lt-5

jsf9k committed:
Install terraform and packer for the linting job

We should be doing this because the Packer and Terraform pre-commit
hooks leverage the corresponding executables; therefore, it makes
sense to go ahead and install the particular versions of those
executables that we support.  Also add support for optionally
debugging via tmate.

See also cisagov/skeleton-generic#74.

jsf9k committed:
Break out the curl cache creation into its own step

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove unnecessary line in tasks

There is no reason to create /usr/bin/terraform.  This is a vestige of
an earlier age.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Modify the Packer installation to model that of Terraform

The Terraform installation does not destroy the existing system
Terraform installation, and neither should the Packer installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Install a specific version of terraform-docs

Note that this change is dependent on the merging of
cisagov/setup-env-github-action#31.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Move go installation so that it takes place before the cache task

Some variables defined in the go installation are used in the cache
task, so the go installation must happen first.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Capitalize Go for consistency

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Update pre-commit hooks

This is performed by running `pre-commit autoupdate`, but with the
`ansible-lint` hook held back manually.

jsf9k committed:
Prefer the newer "go install" syntax

As of [Go 1.16](https://tip.golang.org/doc/go1.16#go-command) the `GO111MODULE` environment variable defaults to `on` and `go get` has been deprecated for module installation.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Install the shfmt tool for GHA

The `shfmt` tool does not ship on the GitHub Actions runners so we must install
it manually.

mcdonnnj committed:
Replace the beautysh hook with pre-commit-shfmt

We have had a difficult time with how beautysh parses some shellscripts. I went
in pursuit of an alternative and I believe shfmt to be a good alternative.

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Apply changes from the shfmt pre-commit hook

mcdonnnj committed:
Merge pull request #83 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit Hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/add_pre-commit_hook

mcdonnnj committed:
Merge pull request #80 from cisagov/improvement/add_pre-commit_hook

Add the validate_manifest Hook to pre-commit Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/update_mdl_configuration

jsf9k committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/update_mdl_configuration

Update markdownlint Configuration

mcdonnnj committed:
Merge branch 'develop' into improvement/install-tf-and-packer-for-linting

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/install-tf-and-packer-for-linting

Install terraform and packer for the linting job

mcdonnnj committed:
Merge branch 'develop' into improvement/replace_beautysh_with_shfmt

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/replace_beautysh_with_shfmt

Replace beautysh hook with cisagov/pre-commit-shfmt

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #78 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/standardize_lineage_configuration

mcdonnnj committed:
Merge pull request #77 from cisagov/improvement/standardize_lineage_configuration

Standardize the Layout of the Lineage Configuration File

mcdonnnj committed:
Merge branch 'develop' into dependabot/pip/ansible-gte-2.10-and-lt-5

mcdonnnj committed:
Merge pull request #75 from cisagov/dependabot/pip/ansible-gte-2.10-and-lt-5

Update ansible requirement from <3,>=2.10 to >=2.10,<5

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into lineage/skeleton

jsf9k committed:
Change unordered list style to make markdownlint happy

dav3r committed:
Fix a typo

jsf9k committed:
Use a matrix strategy for running multiple Molecule scenarios

By default we only have the single default scenario, but there are
some cisagov/ansible-role-* repos where we run multiple scenarios.  In
those cases it makes sense to run them in a matrix since they will run
in parallel; hence it makes sense to go ahead and add the scaffolding
to support that, so folks handle multiple scenarios correctly going
forward.

jsf9k committed:
Merge pull request #21 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

jsf9k committed:
Adjust pre-requisite language to be less COOL-centric

@mcdonnnj pointed out here that such language is not appropriate for
many of our existing Ansible roles:
https://github.com/cisagov/ansible-role-geoip2/pull/8#discussion_r683976710

It is also the case for many of our Ansible roles that, although we
use them exclusively in the COOL, they could be applied elsewhere.

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

mcdonnnj committed:
Use the hashicorp/setup-terraform Action

Instead of manually installing a Terraform binary we can leverage the Action
provided by Hashicorp to do the same thing.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Add a markdownlint rule for horizontal rules

This sets a specific rule for MD035 (Horizontal rule style) instead of the
default value of "consistent".

mcdonnnj committed:
Add a markdownlint rule for code blocks

This sets a specific rule for MD046 (code block style) instead of the default
value of "consistent".

mcdonnnj committed:
Merge branch 'develop' into dav3r-patch-1

mcdonnnj committed:
Merge pull request #85 from cisagov/dav3r-patch-1

Fix a typo

mcdonnnj committed:
Merge branch 'develop' into improvement/specify_mdl_rules

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/specify_mdl_rules

Add markdownlint Rules to Enforce Specific Formatting

mcdonnnj committed:
Merge branch 'develop' into improvement/use_setup-terraform_in_gha_workflow

mcdonnnj committed:
Merge pull request #88 from cisagov/improvement/use_setup-terraform_in_gha_workflow

Use the hashicorp/setup-terraform Action to Install Terraform in GHA Workflow

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Add Fedora 34 as a supported platform

With the release of Fedora 34 on 2021-04-27 it became the latest stable version
of Fedora. Since it is the latest stable release we should have it supported as
one of our platforms for Ansible roles.

mcdonnnj committed:
Use the geerlingguy/docker-debian11-ansible image

Now that geerlingguy/docker-debian11-ansible exists we should use it instead of
cisagov/docker-debian11-ansible. This will bring this platform in line with
other systemd images we use that have available geerlingguy images.

mcdonnnj committed:
Add a table for role variables to the README

Add a sample template for a role variable table to the README. It is commented
out by default, but in projects with role variables it should be populated and
uncommented.

mcdonnnj committed:
Merge pull request #80 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into improvement/use-scenarios-in-a-matrix

mcdonnnj committed:
Merge pull request #79 from cisagov/improvement/use-scenarios-in-a-matrix

Use a matrix strategy for running single or multiple Molecule scenarios

mcdonnnj committed:
Merge branch 'develop' into improvement/use_geerlingguy_debian11_image

mcdonnnj committed:
Merge pull request #81 from cisagov/improvement/use_geerlingguy_debian11_image

Use the geerlingguy/docker-debian11-ansible Image

mcdonnnj committed:
Merge branch 'develop' into improvement/use_table_for_role_variables

mcdonnnj committed:
Merge pull request #84 from cisagov/improvement/use_table_for_role_variables

Add a Table for Role Variables to the README

mcdonnnj committed:
Merge branch 'develop' into improvement/add_fedora34_support

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #82 from cisagov/improvement/add_fedora34_support

Add Fedora 34 as a Supported Platform

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #23 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into documentation/remove-cool-specific-language

mcdonnnj committed:
Merge pull request #22 from cisagov/documentation/remove-cool-specific-language

Adjust pre-requisite language to be less COOL-centric

dav3r committed:
Enable terraform validate pre-commit hook

This hook should now work as expected in most, if not all of our repos 
now that we are finally updating to terraform 0.13.x (on our way to 
1.0.x).

dav3r committed:
Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when testing is completed.

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Revert "Temporarily use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit 6a7fbf07bd371d0493c523ce24647e5c04c77c03.

dav3r committed:
Enable terraform validate pre-commit hook

This hook should work as expected now that we are updating to terraform 
0.13.x (on our way to 1.0.x).

dav3r committed:
Use cisagov/setup-env-github-action@improvement/support_tf_0.13

This change will be reverted when all of our testing is completed.

dav3r committed:
Remove lint job step to initialize Terraform directories

Initialization will now be done during the "terraform validate" step.

dav3r committed:
Bump required_version to ~> 0.13.0

Also, include aws provider source, which was automatically added by 
running the "terraform 0.13upgrade" command.

dav3r committed:
Merge pull request #24 from cisagov/improvement/support_tf_0.13

Support Terraform 0.13

jsf9k committed:
Remove pin for ansible-lint

We can now move to version 5 with impunity.  I have a path forward,
largely following in the footsteps of @geerlingguy.

jsf9k committed:
Add a cisagov namespace to the metadata file

This gets rid of an error from ansible-lint.

jsf9k committed:
Pin ansible-lint to major version 5

@mcdonnnj pointed out that this will keep us from being blindsided by future changes.

Co-authored-by: Nick M. <50747025+mcdonnnj@users.noreply.github.com>

dav3r committed:
Revert "Use cisagov/setup-env-github-action@improvement/support_tf_0.13"

This reverts commit c7f1201ea470ab27f34d6c8584e187ed9e64e890 now
that all Terraform 0.13 testing has been completed.

dav3r committed:
Merge pull request #90 from cisagov/improvement/enable_terraform_validate

Enable terraform validate pre-commit hook

dav3r committed:
Merge pull request #25 from cisagov/improvement/finalize_tf_0.13

Revert commit for TF 0.13 testing branch of cisagov/setup-env-github-action

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

dav3r committed:
Merge pull request #86 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

dav3r committed:
Merge pull request #26 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

dav3r committed:
Use cisagov/setup-env-github-action@improvement/support_tf_0.14

This change will be reverted when all of our testing is completed.

dav3r committed:
Bump required Terraform version to ~> 0.14.0

dav3r committed:
Ignore the Terraform provider dependency lock file

We may commit this file in the future, but for now, we are not ready to 
make this change.

dav3r committed:
Merge pull request #27 from cisagov/improvement/support_tf_0.14

Support Terraform 0.14

dav3r committed:
Revert "Use cisagov/setup-env-github-action@improvement/support_tf_0.14"

This reverts commit 435ee1ccf1bff0509c0484a4cabf74bde8f88829 now
that all Terraform 0.14 testing has been completed.

dav3r committed:
Add cisagov namespace to avoid a molecule error

ERROR    Computed fully qualified role name of skeleton_with_test_user 
does not follow current galaxy requirements.
Please edit meta/main.yml and assure we can correctly determine full 
role name:

galaxy_info:
role_name: my_name  # if absent directory name hosting role is used 
instead
namespace: my_galaxy_namespace  # if absent, author is used instead

Namespace: 
https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations
Role: 
https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names

dav3r committed:
Merge pull request #28 from cisagov/improvement/finalize_tf_0.14

Revert commit for TF 0.14 testing branch of cisagov/setup-env-github-action

dav3r committed:
Use cisagov/setup-env-github-action@improvement/support_tf_1.0

This change will be reverted when all of our testing is completed.

dav3r committed:
Bump required Terraform version to ~> 1.0

dav3r committed:
Merge pull request #29 from cisagov/improvement/support_tf_1.0

Support Terraform 1.0

dav3r committed:
Revert "Use cisagov/setup-env-github-action@improvement/support_tf_1.0"

This reverts commit 26e48ef917f984ee8b74f26a14e7435e2ea68d76
now that all Terraform 1.0 testing has been completed.

dav3r committed:
Merge pull request #30 from cisagov/improvement/finalize_tf_1.0

Revert commit for TF 1.0 testing branch of cisagov/setup-env-github-action

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with switching to v5.

mcdonnnj committed:
Use an id when using cisagov/setup-env-github-action

mcdonnnj committed:
Use setup-env outputs for Terraform version

Change from using an environment variable to the outputs of the
cisagov/setup-env-github-action instead.

mcdonnnj committed:
Use setup-env outputs for Packer version

We use a `PACKER_VERSION` environment variable for the "Install Packer" step
that is populated from the `setup-env` outputs to get around `yamllint` lint
length limits.

mcdonnnj committed:
Use setup-env outputs for shfmt version

We change the "Install shfmt" step to use two local environment variables to
provide the package's URL and version to install. This allows us to work around
`yamllint` line length limits.

mcdonnnj committed:
Use setup-env outputs for Terraform-docs version

We change the "Install Terraform-docs" step to use two local environment
variables to provide the package's URL and version to install. This allows us
to work around `yamllint` line length limits.

mcdonnnj committed:
Store Go version as a step output

This changes from using an environment variable to using a step output to store
the Go version that is installed. This mirrors changes made to the other
program versions and how they're stored.

mcdonnnj committed:
Update the ansible-lint version for pre-commit

mcdonnnj committed:
Add an ansible-lint configuration file

This file is copied from cisagov/skeleton-ansible-role#85 and tweaked to pass
pre-commit. See here for for a list of the elements that can exist in this
file: https://ansible-lint.readthedocs.io/en/latest/configuring.html

Co-authored-by: Shane Frasier <jeremy.frasier@trio.dhs.gov>

mcdonnnj committed:
Add dependabot configuration for Terraform

This will configure `dependabot` to scan Terraform configurations if they exist
in a repository. If a repository stores a Terraform configuration in a sub-
directory this configuration will need to be modified or an additional
configuration must be added if there still exists a Terraform configuration in
the root directory.

mcdonnnj committed:
Merge pull request #91 from cisagov/improvement/pre-commit_update

Update `pre-commit` Hooks

mcdonnnj committed:
Merge pull request #93 from cisagov/improvement/update_ansible-lint_pre-commit_version

Update the `ansible-lint` Version in the pre-commit Configuration

mcdonnnj committed:
Merge pull request #94 from cisagov/improvement/add_dependabot_configuration_for_terraform

Add `dependabot` Configuration for Terraform

mcdonnnj committed:
Merge pull request #92 from cisagov/improvement/use_action_outputs

Use Outputs of cisagov/setup-env-github-action to Control Installed Program Versions

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #87 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #85 from cisagov/improvement/upgrade-ansible-lint

Unpin ansible-lint

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into lineage/skeleton

jsf9k committed:
Configure dependabot to check Terraform code

jsf9k committed:
Merge pull request #31 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Update the molecule converge configuration

Update the format of the `converge.yml` file to mirror the template file
produced when running `molecule init role <role name> --driver-name docker`
with the latest version of molecule. This is also necessary to get non-default
scenarios to play nicely with v5 of ansible-lint.

mcdonnnj committed:
Merge pull request #88 from cisagov/maintenance/update_converge_format

Update the Configuration for the Molecule Converge Step

mcdonnnj committed:
Add Debian 12 (Bookworm) as a platform

Debian Bookworm is the current version of Debian Testing. Since we test against
Kali Linux, and Kali Linux is built on Debian Testing, it makes sense that we
should also test against the current version of Debian Testing.

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into lineage/skeleton

jsf9k committed:
Merge pull request #32 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Update ansible-lint configuration

Remove the manual exclusion of the `converge.yml` file in the default molecule
scenario. With the changes in cisagov/skeleton-ansible-role#88 this exclusion
is no longer necessary.

mcdonnnj committed:
Update ansible-lint configuration

Instead of excluding molecule configurations for systemd individually we
instead direct ansible-lint to process any of these configurations in
scenario directories as plain yaml files. This mirrors how ansible-lint
handles `molecule.yml` files in scenario directories by default.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

dependabot[bot] committed:
Update ansible requirement from <5,>=2.10 to >=2.10,<6

Updates the requirements on [ansible](https://github.com/ansible/ansible) to permit the latest version.
- [Release notes](https://github.com/ansible/ansible/releases)
- [Commits](https://github.com/ansible/ansible/commits)

---
updated-dependencies:
- dependency-name: ansible
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

jsf9k committed:
Change underscores in hostnames to hyphens

Underscores in hostnames are disallowed according to RFC 1035:
https://www.rfc-editor.org/rfc/rfc1035.html

Our use of underscores is currently breaking our Kali-based Ansible
role testing:
https://github.com/cisagov/ansible-role-orchestrator/runs/4644951540?check_suite_focus=true#step:6:266

jsf9k committed:
Merge pull request #92 from cisagov/bugfix/no-underscores-in-hostnames

Change underscores in hostnames to hyphens

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #97 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Merge branch 'develop' into improvement/update_ansible-lint_configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/update_ansible-lint_configuration

Update `ansible-lint` Configuration

Commit:
Merge https://github.com/cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Merge pull request #93 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge branch 'develop' into dependabot/pip/ansible-gte-2.10-and-lt-6

mcdonnnj committed:
Merge pull request #91 from cisagov/dependabot/pip/ansible-gte-2.10-and-lt-6

Update ansible requirement from <5,>=2.10 to >=2.10,<6

mcdonnnj committed:
Merge pull request #89 from cisagov/improvement/add_debian12_configuration

Add Debian 12 (Bookworm) as a Platform

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

jsf9k committed:
Merge pull request #34 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

jsf9k committed:
Correct the hostname that appears in the host_vars section

I changed the hostname above in cisagov/skeleton-ansible-role#92 but
neglected to update it here.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Add Fedora 35, remove Fedora 32 and 33

Fedora 33 and 33 are EOL, so we need not continue to support them.

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`.

mcdonnnj committed:
Add comment to Dependabot configuration

This comment explains that the configuration may have commented out
ignore directives that should be uncommented in downstream projects.

mcdonnnj committed:
Disable comments-indentation rule for yamllint

yamllint does not like it when you comment out pieces of dictionaries
in lists. Upcoming additions to the Dependabot configuration will run
afoul of this so we are updating the yamllint configuration.

mcdonnnj committed:
Merge pull request #99 from cisagov/improvement/update_dependabot_and_yamllint_configurations

Add boilerplate for upcoming Dependabot configuration changes

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #98 from cisagov/maintenance/update_pre-commit_hooks

Update pre-commit hooks

mcdonnnj committed:
Use asterisks for emphasis- and strong-styles

Given the inconsistent behavior of underscores used within words for
style we should prefer asterisks now that these rules are available.

mcdonnnj committed:
Add missing hyphen in markdownlint rule header

mcdonnnj committed:
Merge pull request #100 from cisagov/improvement/update_markdownlint_configuration

Use new `markdownlint` rules for emphasis- and strong-styles

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for ansible and ansible-lint.
These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

dependabot[bot] committed:
Bump actions/setup-python from 2 to 3

Bumps [actions/setup-python](https://github.com/actions/setup-python) from 2 to 3.
- [Release notes](https://github.com/actions/setup-python/releases)
- [Commits](https://github.com/actions/setup-python/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/setup-python
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 2 to 3

Bumps [actions/checkout](https://github.com/actions/checkout) from 2 to 3.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge pull request #96 from cisagov/lineage/skeleton

⚠️ CONFLICT! Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #97 from cisagov/improvement/add_dependabot_ignore_directives

Add Dependabot ignore directives for `ansible` and `ansible-lint`

mcdonnnj committed:
Merge branch 'develop' into improvement/remove-eol-distros-add-new-distros

mcdonnnj committed:
Merge pull request #95 from cisagov/improvement/remove-eol-distros-add-new-distros

Add Fedora 35, remove Fedora 32 and 33

mcdonnnj committed:
Merge branch 'develop' into bugfix/correct-hostname

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #94 from cisagov/bugfix/correct-hostname

Correct the hostname that appears in the host_vars section

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Enable Dependabot ignore directive

This enables the Dependabot ignore directive for pip in the root
directory.

mcdonnnj committed:
Add Dependabot ignore for hashicorp/aws

This ignore is commented out in the skeleton but should be uncommented
in downstream projects so that they only get updates for this
dependency from the skeleton.

mcdonnnj committed:
Merge pull request #41 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

mcdonnnj committed:
Merge pull request #42 from cisagov/improvement/update_dependabot_configuration

Add Dependabot ignore directive

mcdonnnj committed:
Update pre-commit hooks

Update pre-commit hooks using `pre-commit autoupdate`. The `ansible-lint` hook
is intentionally held back due to issues with upgrading to v6.

mcdonnnj committed:
Add Dependabot ignore directives

This adds commented out ignore directives for the following GitHub
Actions:

- action/cache
- action/checkout
- action/setup-python

These should be uncommented downstream to ensure that updates to these
dependencies are pushed from pull requests made in the skeleton.

mcdonnnj committed:
Upgrade from Python 3.9 to 3.10 for the `lint` job in GHA

mcdonnnj committed:
Use consistent quoting for software versions

mcdonnnj committed:
Merge pull request #101 from cisagov/dependabot/github_actions/actions/setup-python-3

Bump actions/setup-python from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into dependabot/github_actions/actions/checkout-3

mcdonnnj committed:
Merge pull request #102 from cisagov/dependabot/github_actions/actions/checkout-3

Bump actions/checkout from 2 to 3

dependabot[bot] committed:
Bump actions/cache from 2 to 3

Bumps [actions/cache](https://github.com/actions/cache) from 2 to 3.
- [Release notes](https://github.com/actions/cache/releases)
- [Commits](https://github.com/actions/cache/compare/v2...v3)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

mcdonnnj committed:
Merge branch 'develop' into improvement/add_dependabot_ignores_for_github_actions

mcdonnnj committed:
Merge pull request #107 from cisagov/improvement/add_dependabot_ignores_for_github_actions

Add Dependabot ignore directives

mcdonnnj committed:
Merge pull request #104 from cisagov/dependabot/github_actions/actions/cache-3

Bump actions/cache from 2 to 3

mcdonnnj committed:
Merge branch 'develop' into improvement/use_python_3.10_for_gha

mcdonnnj committed:
Merge pull request #106 from cisagov/improvement/use_python_3.10_for_gha

Use Python 3.10 for the `lint` job in GitHub Actions

mcdonnnj committed:
Merge branch 'develop' into maintenance/update_pre-commit_hooks

mcdonnnj committed:
Merge pull request #105 from cisagov/maintenance/update_pre-commit_hooks

Update `pre-commit` hooks

mcdonnnj committed:
Add pre-commit hooks to validate GitHub Actions content

Add hooks to validate GitHub Actions and GitHub Action workflows. These
are validated using JSON Schemas.

mcdonnnj committed:
Merge pull request #109 from cisagov/improvement/add_github_actions_pre-commit_hooks

Add `pre-commit` hooks to validate GitHub Actions content

mcdonnnj committed:
Merge github.com:cisagov/skeleton-generic into lineage/skeleton

mcdonnnj committed:
Enable Dependabot ignore directives

This enables the Dependabot ignore directives for GitHub Actions.

mcdonnnj committed:
Merge pull request #103 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

Commit:
Merge https://github.com/cisagov/skeleton-ansible-role into lineage/skeleton

mcdonnnj committed:
Merge pull request #48 from cisagov/lineage/skeleton

Lineage pull request for: skeleton

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role-with-test-user/develop' into skeletonize

jsf9k committed:
Update lineage configuration

jsf9k committed:
Update codeowners to only those knowledgeable about this repo

jsf9k committed:
Fix TF code formatting

jsf9k committed:
Use FQCNs everywhere

Also update code to handle various ansible-lint complaints.

jsf9k committed:
Do not needlessly install the pip role a second time

jsf9k committed:
Remove unnecessary Kali vars file and add RedHat vars file

jsf9k committed:
Fix bug where has_latest was not being treated as a boolean value

jsf9k committed:
Remove unsupported platforms

jsf9k committed:
Replace skeleton boilerplate in role metadata

jsf9k committed:
Fix when clause in task

For some reason the when clause didn't work correctly when I set a
fact.  Now that the fact is only being used in one place, it makes
sense to do away with it altogether anyway.

dav3r commented on .github/dependabot.yml:
```diff
@@ -0,0 +1,32 @@
+    schedule:
+      interval: "weekly"
+    # ignore:
+    #   - dependency-name: "hashicorp/aws"
```
We want to uncomment this bit inherited from `skeleton-ansible-role-with-test-user`:
```suggestion
    ignore:
      - dependency-name: "hashicorp/aws"
```

dav3r commented on README.md:
```diff
@@ -7,18 +7,66 @@
+|----------|-------------|---------|----------|
+| package_bucket | The name of the AWS S3 bucket containing the Nessus Debian package file. | `cisa-cool-third-party-production` | No |
+| package_file | The name of the Nessus Debian package file. | `Nessus-{{ version }}-debian6_amd64.deb` | No |
+| version | The version number of the Nessus Debian package file. | `8.7.1` | No |
```
I was speaking with @mcdonnnj and he mentioned that he thought you would be updating us to the newest version of Nessus 8.x (currently `8.15.4`).  I just wanted to confirm if you planned to do that in a separate PR or if you were going to include it in this one.

jsf9k replied:
I created #16, which failed automated testing; that's how I realized that I needed to re-skeletonize.

dav3r replied:
Thanks- I should've checked myself (hence why I wrecked myself).

dav3r commented on terraform/providers.tf:
```diff
@@ -3,37 +3,77 @@
+# everything the CI user needs to do in the staging account.
 provider "aws" {
-  alias  = "images-production"
+  alias = "images_staging_provisionaccount"
```
Can you please re-order these providers alphabetically so that `aws.images_production_provisionaccount` comes before `aws.images_staging_provisionaccount`?

jsf9k replied:
Please see commit 99bfd75.

dav3r commented on terraform/variables.tf:
```diff
@@ -30,7 +31,7 @@ variable "tags" {
-    Team        = "NCATS OIS - Development"
-    Application = "ansible-role-nessus testing"
+    Team        = "VM Fusion - Development"
+    Application = "skeleton-ansible-role-with-test-user testing"
```
Looks like you accidentally overwrote this:
```suggestion
    Application = "ansible-role-nessus testing"
```

dav3r reviewed (changes requested):
Good stuff, though I spotted a few thangs you missed.

Thanks for being a re-skeletonizer! 💀

jsf9k committed:
Correct application tag

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Uncomment dependabot ignore directive

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Correctly alphabetize providers by alias

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

dav3r reviewed (approved):
Outstanding! 🤖

mcdonnnj commented on meta/main.yml:
```diff
@@ -3,18 +3,26 @@ galaxy_info:
-  galaxy_tags:
-    - nessus
+        - bookworm
+  role_name: nessus
```
Since these platforms are in the molecule configuration now.

```suggestion
    - name: Ubuntu
      versions:
        - bionic
        - focal
  role_name: nessus
```

mcdonnnj commented on molecule/default/prepare.yml:
```diff
@@ -1,10 +1,15 @@
+
+# boto3 is pre-installed anywhere we would be applying this Ansible
+# role
+- name: Install boto3 and pip
```
Since `pip` is installed in the `python.yml` playbook above.

```suggestion
- name: Install boto3
```

mcdonnnj reviewed (changes requested):
Thanks for tackling another one of these lost lambs. In addition to my suggestions below I believe we can now remove the `regex_replace()` logic in the following:
https://github.com/cisagov/ansible-role-nessus/blob/99bfd7572032a2d7ccefd612df19d7a72b4bc89e/tasks/main.yml#L52-L57

Here's a reference to a commit doing so in another Ansible role: https://github.com/cisagov/ansible-role-clamav/commit/8b0fb1792011089b013cb4add3f31de40be8cd68

jsf9k committed:
Add Ubuntu platforms to role metadata

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove redundant installation of pip role

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Remove regex_replace() foo

This foo is no longer necessary, since Ansible now correctly
identifies Kali Linux as being the Kali distribution of Debian.

Also prefer ansible_distribution_release to
ansible_distribution_version.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

jsf9k commented:
> Thanks for tackling another one of these lost lambs. In addition to my suggestions below I believe we can now remove the `regex_replace()` logic in the following:
> 
> https://github.com/cisagov/ansible-role-nessus/blob/99bfd7572032a2d7ccefd612df19d7a72b4bc89e/tasks/main.yml#L52-L57
> 
> Here's a reference to a commit doing so in another Ansible role: [cisagov/ansible-role-clamav@8b0fb17](https://github.com/cisagov/ansible-role-clamav/commit/8b0fb1792011089b013cb4add3f31de40be8cd68)

Good call!

mcdonnnj reviewed (approved):
:shipit:
